internal/statestore/public.go

```diff
@@ -28,34 +28,34 @@ type Service interface {
 	HealthCheck(ctx context.Context) error
 
 	// CreateTicket creates a new Ticket in the state storage. If the id already exists, it will be overwritten.
-	CreateTicket(ctx context.Context, ticket *pb.Ticket) error
+	CreateTicket(ctx context.Context, ticket *pb.Ticket) (context.Context, error)
```

Comment:
I dont think returning the context from state-store APIs is the right pattern. Is this just so you can have the instrumented context outside of this API? If so, then the instrumentation should be done earlier in the stack than passing the instrumented context back out from this.

Reply:
Yea it seems like grpc ecosystem provides a tracing middleware. I’ll try that out and see if it could improve the context propagation